src/plot_api/subroutines.js

```diff
@@ -397,24 +399,98 @@ function findCounterAxisLineWidth(ax, side, counterAx, axList) {
 }
 
 exports.drawMainTitle = function(gd) {
+    var title = gd._fullLayout.title;
+    setDflts(title);
     var fullLayout = gd._fullLayout;
-
     var textAnchor = getMainTitleTextAnchor(fullLayout);
     var dy = getMainTitleDy(fullLayout);
+    var y = getMainTitleY(fullLayout, dy);
+    var position = fullLayout.title.y > 0.5 ? 't' : 'b';
+
+    if(gd._fullLayout.title.automargin) {
+        applyTitleAutoMargin(gd, position, y);
+    }
 
     Titles.draw(gd, 'gtitle', {
         propContainer: fullLayout,
         propName: 'title.text',
         placeholder: fullLayout._dfltTitle.plot,
         attributes: {
             x: getMainTitleX(fullLayout, textAnchor),
-            y: getMainTitleY(fullLayout, dy),
+            y: y,
             'text-anchor': textAnchor,
             dy: dy
         }
     });
 };
 
+// Is any part of the title outside of the container?
+// TODO: What if yref='container'?
+function isOutsideContainer(gd, title, position, y) {
+    var plotHeight = gd._fullLayout._size.h;
+    var yPosTop = Lib.isTopAnchor(title) ? y : y - titleDepth(title); // Standardize to the top of the title
+    var yPosRel = position === 'b' ? plotHeight - yPosTop : yPosTop; // Position relative to the top or bottom of plot
+    if((Lib.isTopAnchor(title) && position === 't') || Lib.isBottomAnchor(title) && position === 'b') {
+        return false;
+    } else {
+        if(yPosRel < titleDepth(title)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+}
+
+// TODO: Move to setting defaults stage, rather than drawing
+// title.y is 1 or 0 if automargin and paper ref
```

Comment:
Yes we should move this part to supply defaults step.
I feel the `y` position should not be defined when `automargin` is *true*.
And instead of `y` values to position the title on top or at the bottom, we could/should use `yanchor` values.
Does this make sense?
Also let's add a separate mock to test the positioning of the title at the bottom.
Thank you!

Reply:
Yes, I'll add in the mock! Using `yanchor` to control top of bottom position would be interesting... Although I think this would add complexity for the `container` positioning as then the `y` is relevant for pushing the size of the margin (something that I also need to add in tests around).

Reply:
Yes, this logic should all move into supplyDefaults.

I still don't understand why we're restricting `title.y` to either 0 or 1. Yes, 99% of the time that's what people will use, but if someone gets into the details of tweaking their graph and wants to use `title.y` to move the title a little up or down why would we prevent this? Worse still, if they've forgotten that they turned on `title.automargin` and they try to adjust `title.y` from 0 to 0.01 and it snaps all the way to the top, they'll have to somehow figure out that they need to turn off `title.automargin` before they can get the desired effect?

Reply:
Agreeing to leave `title.y` as 0 or 1 in this case, and add a console warning when it is being reverted to these values.